src/document/transformoperation.js

```diff
@@ -0,0 +1,434 @@
+ * operations were applied. This way all modifications done to the Tree Model will be reflected
+ * in the operation parameters and the operation will "operate" on "up-to-date" version of the Tree Model.
+ * This is mostly the case with Operational Transformation but it might be needed in particular features.
+ *
+ * In some cases, when given operation apply changes to the same nodes as this operation, there is a need
+ * to create two operations as a result. It would be impossible to create just one operation that handles
+ * modifications needed to be applied to the tree. This is why Array is returned instead of single object.
+ * All returned operations has to be applied (or further transformed) to get an effect that was intended in
+ * pre-transformed operation.
+ *
+ * @function transformOperation
+ * @param {document.operation.Operation} a Operation that will be transformed.
+ * @param {document.operation.Operation} b Operation to transform by.
+ * @param {Boolean} isStrong Flag indicating whether this operation should be treated as more important
+ * when resolving conflicts.
+ * @returns {Array.<document.operation.Operation>} Result of the transformation.
+ */
+
+CKEDITOR.define( [
+	'document/operation/insertoperation',
+	'document/operation/changeoperation',
+	'document/operation/moveoperation',
+	'document/operation/nooperation',
+	'document/range',
+	'utils'
+], ( InsertOperation, ChangeOperation, MoveOperation, NoOperation, Range, utils ) => {
+
+	// When we don't want to update an operation, we create and return a clone of it.
+	// Returns the operation in "unified format" - wrapped in an Array.
+	function doNotUpdate( operation ) {
+		return [ operation.clone() ];
+	}
+
+	// Takes an Array of operations and sets consecutive base versions for them, starting from given base version.
+	// Returns the passed array.
+	function updateBaseVersions( baseVersion, operations ) {
+		for ( let i = 0; i < operations.length; i++ ) {
+			operations[ i ].baseVersion = baseVersion + i + 1;
+		}
+
+		return operations;
+	}
+
+	// Checks whether MoveOperation targetPosition is inside a node from the moved range of the other MoveOperation.
+	function moveTargetIntoMovedRange( a, b ) {
+		return a.targetPosition.getTransformedByDeletion( b.sourcePosition, b.howMany ) === null;
+	}
+
+	const ot = {
+		InsertOperation: {
+			/**
+			 * Returns an array containing the result of transforming
+			 * {document.operation.InsertOperation} by {document.operation.InsertOperation}.
+			 *
+			 * @param {document.operation.InsertOperation} a Operation that will be transformed.
+			 * @param {document.operation.InsertOperation} b Operation to transform by.
+			 * @param {Boolean} isStrong Flag indicating whether this operation should be treated as more important
+			 * when resolving conflicts.
+			 * @returns {Array.<document.operation.InsertOperation>} Result of the transformation.
+			 */
```

Comment:
So, basically, earlier these functions were parts of classes so I JSDoc'ed them. Also the docs were a bit more different. I didn't want to delete these comments, but you are right, now it doesn't have much sense to keep them in this state.